smithy-trait-codegen/codegen-core/src/main/java/software/amazon/smithy/traitcodegen/generators/traits/EnumTraitGenerator.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.traitcodegen.generators.traits;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.model.FromSourceLocation;
+import software.amazon.smithy.model.SourceLocation;
+import software.amazon.smithy.model.shapes.EnumShape;
+import software.amazon.smithy.model.traits.StringTrait;
+import software.amazon.smithy.traitcodegen.GenerateTraitDirective;
+import software.amazon.smithy.traitcodegen.generators.common.PropertiesGenerator;
+import software.amazon.smithy.traitcodegen.writer.TraitCodegenWriter;
+import software.amazon.smithy.utils.StringUtils;
+
+final class EnumTraitGenerator extends TraitGenerator {
```

Comment:
Why don't enum traits just return their enum? You wouldn't need things like the `is` methods. Same for int enums.

Reply:
I use strings for enum traits for two reasons: 
1. String Enums can inherit from StringTrait
2. The Enums stay consistent with our existing enum trait implementations (examples: [data](https://github.com/smithy-lang/smithy/blob/main/smithy-aws-traits/src/main/java/software/amazon/smithy/aws/traits/DataTrait.java), [cfnMutability](https://github.com/smithy-lang/smithy/blob/main/smithy-aws-cloudformation-traits/src/main/java/software/amazon/smithy/aws/cloudformation/traits/CfnMutabilityTrait.java)). This only really applies for String Enums as I dont think there are any instances of IntEnum Traits in the main repo.

Reply:
1. `getEnumValue`
2. We can always do better